src/wasi.ts

```diff
@@ -1,33 +1,27 @@
 import * as wasi from "./wasi_defs.js";
 import { Fd } from "./fd.js";
 
-type mixed = any;
-
-declare var TextEncoder: {
-  prototype: TextEncoder;
-  new (encoding?: string): TextEncoder;
-};
+const DUMMY = -1;
 
 export default class WASI {
   args: Array<string> = [];
   env: Array<string> = [];
   fds: Array<Fd> = [];
-  //@ts-ignore
   inst: { exports: { memory: WebAssembly.Memory } };
-  wasiImport: { [key: string]: (...args: Array<any>) => mixed };
+  wasiImport: { [key: string]: (...args: Array<unknown>) => unknown };
```

Comment:
I'm not all that familiar with typescript myself. What is exactly the difference between any and unknown?

Reply:
Unknown is essentially an `any` type that can't be assigned to or used without narrowing the type.

`any` is like a shortcut out of TypeScript type checking. You can do anything/call anything on an `any` type. 

`unknown` is like telling TypeScript you don't know or care what this type is yet. Maybe you're just propagating an `unknown` value to another function, you're not using the `unknown` value, or you're deferring type handling to a later time.

`unknown` was introduced because `any` was too loose for the myriad second cases. Once it was introduced, `any` became a common lint error/warning.